Source/Diagnostics/ReducedDiags/BeamRelevant.cpp

```diff
@@ -0,0 +1,376 @@
+
+        // get WarpXParticleContainer class object
+        auto & myspc = mypc.GetParticleContainer(i_s);
+
+        // get mass (Real)
+        auto m = myspc.getMass();
+
+        using PType = typename WarpXParticleContainer::SuperParticleType;
+
+        // weight sum
+        auto w_sum = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::w); });
+
+        // reduced sum over mpi ranks
+        ParallelDescriptor::ReduceRealSum(w_sum);
+
+        if (w_sum < std::numeric_limits<Real>::min() )
+        {
+            for (int i = 0; i < m_data.size(); ++i) { m_data[i] = 0.0; }
+            return;
+        }
+
+        // x mean
+        auto x_mean = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.pos(0) * p.rdata(PIdx::w) / w_sum; });
+
+        #if (AMREX_SPACEDIM == 3)
+        // y mean
+        auto y_mean = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.pos(1) * p.rdata(PIdx::w) / w_sum; });
+        #endif
+
+        // z mean
+        auto z_mean = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.pos(index_z) * p.rdata(PIdx::w) / w_sum; });
+
+        // ux mean
+        auto ux_mean = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::ux) * p.rdata(PIdx::w) / w_sum; });
+
+        // uy mean
+        auto uy_mean = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::uy) * p.rdata(PIdx::w) / w_sum; });
+
+        // uz mean
+        auto uz_mean = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        { return p.rdata(PIdx::uz) * p.rdata(PIdx::w) / w_sum; });
+
+        // gamma mean
+        auto gm_mean = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            auto ux = p.rdata(PIdx::ux);
+            auto uy = p.rdata(PIdx::uy);
+            auto uz = p.rdata(PIdx::uz);
+            auto us = ux*ux + uy*uy + uz*uz;
+            return std::sqrt(1.0 + us*inv_c2) * p.rdata(PIdx::w) / w_sum;
+        });
+
+        // reduced sum over mpi ranks
+        ParallelDescriptor::ReduceRealSum(x_mean);
+        #if (AMREX_SPACEDIM == 3)
+        ParallelDescriptor::ReduceRealSum(y_mean);
+        #endif
+        ParallelDescriptor::ReduceRealSum(z_mean);
+        ParallelDescriptor::ReduceRealSum(ux_mean);
+        ParallelDescriptor::ReduceRealSum(uy_mean);
+        ParallelDescriptor::ReduceRealSum(uz_mean);
+        ParallelDescriptor::ReduceRealSum(gm_mean);
+
+        // x rms
+        auto x_rms = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            auto a = (p.pos(0)-x_mean) * (p.pos(0)-x_mean);
+            return a * p.rdata(PIdx::w) / w_sum;
+        });
```

Comment:
The name is confusing: this is *not* RMS, because you don't take the root here. Could you give this variable (and others similar) a more consistent name, for instance `x_mean_square`?

Reply:
Did that.